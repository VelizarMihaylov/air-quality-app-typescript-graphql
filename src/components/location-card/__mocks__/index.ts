export const measurements = [
  {
    __typename: 'Measurement',
    parameter: 'o3',
    value: 25,
    unit: 'µg/m³'
  },
  {
    __typename: 'Measurement',
    parameter: 'so2',
    value: 12,
    unit: 'µg/m³'
  },
  {
    __typename: 'Measurement',
    parameter: 'pm25',
    value: 6,
    unit: 'µg/m³'
  },
  {
    __typename: 'Measurement',
    parameter: 'no2',
    value: 16,
    unit: 'µg/m³'
  }
]
