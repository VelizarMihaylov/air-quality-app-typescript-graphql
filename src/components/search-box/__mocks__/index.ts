export const cities = [
  {
    name: 'Manchester'
  },
  {
    name: 'London'
  },
  {
    name: 'Liverpool'
  },
  {
    name: 'Birmingham'
  }
]
