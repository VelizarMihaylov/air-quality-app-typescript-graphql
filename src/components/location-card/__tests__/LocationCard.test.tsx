import React from 'react'
import { create } from 'react-test-renderer'

import { LocationCard } from '..'

import { ThemeProvider } from 'styled-components'
import generateTheme from 'theme'

import dayjs from 'dayjs'
jest.mock('dayjs')
const mockedDayJs: jest.Mock<unknown> = dayjs as never

import { measurements } from '../__mocks__'

describe('LocationCard', () => {
  it('should render with city, location and measurement props', () => {
    mockedDayJs.mockImplementation(() => ({
      extend: jest.fn(),
      utc: (): { fromNow: () => string } => ({
        fromNow: (): string => '5 hours ago'
      })
    }))
    const LocationCardRender = create(
      <ThemeProvider theme={generateTheme()}>
        <LocationCard
          city="Manchester"
          location="Manchester Piccadilly"
          measurements={measurements}
          lastUpdated="2019-12-28T08:00:00.000Z"
        />
      </ThemeProvider>
    ).toJSON()
    expect(LocationCardRender).toMatchSnapshot()
  })
})
