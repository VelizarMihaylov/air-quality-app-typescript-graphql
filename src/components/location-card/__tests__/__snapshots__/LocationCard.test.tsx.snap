// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationCard should render with city, location and measurement props 1`] = `
<div
  className="sc-bdfBwQ fSvdqj"
  data-puppet="location-card"
>
  <div
    className="card-content"
  >
    <p
      className="updated-at"
    >
      Updated 5 hours ago
    </p>
    <h3>
      Manchester Piccadilly
    </h3>
    <p>
      in Manchester, United Kingdom
    </p>
    <p
      className="values"
    >
      Value: O3: 25 µg/m³, SO2: 12 µg/m³, PM25: 6 µg/m³, NO2: 16 µg/m³
    </p>
  </div>
</div>
`;
